File: Definitions/Game_Definitions/classes/PersonnelSystem.lua
---@meta
---@diagnostic disable

---@class PersonnelSystem : DeviceSystemBase
---@field EnableE3QuickHacks Bool
PersonnelSystem = {}

---@return PersonnelSystem
function PersonnelSystem.new() return end

---@param props table
---@return PersonnelSystem
function PersonnelSystem.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function PersonnelSystem:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function PersonnelSystem:OnTakeControl(ri) return end

---@return PersonnelSystemController
function PersonnelSystem:GetController() return end

---@return PersonnelSystemControllerPS
function PersonnelSystem:GetDevicePS() return end

File: DecompiledGameScripts/cyberpunk/network/systems/personnelSystem.swift

public class PersonnelSystem extends DeviceSystemBase {

  private let m_EnableE3QuickHacks: Bool;

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as PersonnelSystemController;
    super.OnTakeControl(ri);
  }

  protected const func GetController() -> ref<PersonnelSystemController> {
    return this.m_controller as PersonnelSystemController;
  }

  public const func GetDevicePS() -> ref<PersonnelSystemControllerPS> {
    return this.GetController().GetPS();
  }
}
